File: apps/android/app/src/main/java/com/exponential/app/data/api/SubscriptionsApi.kt
package com.exponential.app.data.api

import javax.inject.Inject
import javax.inject.Singleton
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

// Mirrors apps/web/src/lib/trpc/subscriptions.ts — per-issue subscribe toggle.
@Serializable
private data class SubIssueIdInput(@SerialName("issueId") val issueId: String)

@Singleton
class SubscriptionsApi @Inject constructor(private val trpc: TrpcClient) {

    suspend fun subscribe(accountId: String, issueId: String) {
        trpc.mutationUnit(
            accountId,
            path = "subscriptions.subscribe",
            input = SubIssueIdInput(issueId),
            inputSerializer = SubIssueIdInput.serializer(),
        )
    }

    suspend fun unsubscribe(accountId: String, issueId: String) {
        trpc.mutationUnit(
            accountId,
            path = "subscriptions.unsubscribe",
            input = SubIssueIdInput(issueId),
            inputSerializer = SubIssueIdInput.serializer(),
        )
    }
}

File: apps/ios/ExpCore/Sources/API/SubscriptionsApi.swift
import Foundation

// Mirrors apps/web/src/lib/trpc/subscriptions.ts — per-issue subscribe toggle.
private struct SubIssueIdInput: Encodable {
    let issueId: String
}

private struct EmptyResult: Decodable {}

public final class SubscriptionsApi: Sendable {
    private let trpc: TrpcClient

    public init(trpc: TrpcClient) {
        self.trpc = trpc
    }

    public func subscribe(accountId: String, issueId: String) async throws {
        let _: EmptyResult = try await trpc.mutation(
            accountId: accountId,
            path: "subscriptions.subscribe",
            input: SubIssueIdInput(issueId: issueId)
        )
    }

    public func unsubscribe(accountId: String, issueId: String) async throws {
        let _: EmptyResult = try await trpc.mutation(
            accountId: accountId,
            path: "subscriptions.unsubscribe",
            input: SubIssueIdInput(issueId: issueId)
        )
    }
}
